exec/cnex/cnex.c

```diff
@@ -28,28 +39,41 @@ typedef struct tagTCommandLineOptions {
     int  ArgStart;
     char *pszFilename;
     char *pszExecutableName;
+    char *pszExecutablePath;
 } TOptions;
 
-static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL };
+static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL, NULL };
+
 
 void exec_freeExecutor(TExecutor *e)
 {
-    uint32_t i;
-
-    for (i = 0; i < e->object->global_size; i++) {
-        cell_clearCell(&e->globals[i]);
-    }
-    free(e->globals);
+    // First, assert that we've emptied the stack during normal execution.
     assert(isEmpty(e->stack));
     destroyStack(e->stack);
     free(e->stack);
+
+    // Next, iterate all allocated modules, freeing them as we go.
+    // Note that we do not free the last (main) module until all others are freed.
+    for (unsigned int m = 1; m < e->module_count; m++) {
```

Comment:
Perhaps it would be cleaner to run this loop backwards, so that you free module[0] last as part of the loop, rather than having a separate exception just for the main module.

Reply:
Fixed.